xtevent/_eventols.ado

```diff
@@ -159,17 +170,98 @@ program define _eventols, rclass
 		}
 	}
 	else loc indepvars ""
+		
+	******** SA *******
+	if "`sun_abraham'"!=""{
+		* Parse the dependent variable
+		local lhs = "`depenvar'"
+		local rel_time_list = "`included'"
+		
+		* Convert the varlist of relative time indicators to nvarlist 
+		local nvarlist "" // copies of relative time indicators with control cohort set to zero
+		local dvarlist "" // for display
+		foreach l of varlist `rel_time_list' {
+			local dvarlist "`dvarlist' `l'"
+			tempname n`l'
+			qui gen `n`l'' = `l'
+			qui replace `n`l'' = 0 if  `control_cohort' == 1 
+			local nvarlist "`nvarlist' `n`l''" 
+		}	
+
+		* Get cohort count  and count of relative time
+		qui levelsof `cohort' if  `control_cohort' == 0, local(cohort_list) 
+		local nrel_times: word count `nvarlist' 
+		local ncohort: word count `cohort_list'  
+		
+		**** step 2 
+		* Initiate empty matrix for weights 
+		tempname bb ff_w
+		
+		* Loop over cohort and get cohort shares for relative times
+		local nresidlist ""
+		foreach yy of local cohort_list {
+			tempvar cohort_ind resid`yy'
+			qui gen `cohort_ind'  = (`cohort' == `yy') 
+			qui regress `cohort_ind' `nvarlist'  if `touse' & `control_cohort' == 0 [`weight'`exp']  , nocons
+			mat `bb' = e(b) 
+			matrix `ff_w'  = nullmat(`ff_w') \ `bb' 
+			*di "`yy'"
+			*mat li `ff_w'
+			qui predict double `resid`yy'', resid 
+			local nresidlist "`nresidlist' `resid`yy''" //list of variables of residuals 
+		}
+		
+		* Get VCV estimate for the cohort shares using avar
+		* In case users have not set relative time indicators to zero for control cohort
+		* Manually restrict the sample to non-control cohort
+		tempname XX Sxx Sxxi S KSxxi Sigma_ff
+		qui mat accum `XX' = `nvarlist' if  `touse' & `control_cohort' == 0 [`weight'`exp'], nocons
+		mat `Sxx' = `XX'*1/r(N) 
+		mat `Sxxi' = syminv(`Sxx') 
+		qui avar (`nresidlist') (`nvarlist')  if `touse' & `control_cohort' == 0 [`weight'`exp'], nocons robust

```

Comment:
@Constantino-Carreto-Romero Do we have a check for whether `avar` is installed earlier in the code, or in `xtevent.ado` ?

Reply:
I implemented it in https://github.com/JMSLab/xtevent/commit/5b93fc240e456eac1a3d781eaeeaee3e8537bbf3. Then, I ran some examples.